src/plone/restapi/services/aliases/get.py

```diff
@@ -22,38 +23,53 @@ def __init__(self, context, request):
         self.context = context
         self.request = request
 
-    def reply_item(self):
+    def reply(self):
         storage = getUtility(IRedirectionStorage)
+        form = self.request.form
+        portal_path = "/".join(self.context.getPhysicalPath()[:2])
         context_path = "/".join(self.context.getPhysicalPath())
-        redirects = storage.redirects(context_path)
-        aliases = [deroot_path(alias) for alias in redirects]
-        self.request.response.setStatus(200)
-        self.request.response.setHeader("Content-Type", "application/json")
-        return [{"path": alias} for alias in aliases], len(aliases)
-
-    def reply_root(self):
-        """
-        redirect-to - target
-        path        - path
-        redirect    - full path with root
-        """
-        batch = RedirectsControlPanel(self.context, self.request).redirects()
-        redirects = [entry for entry in batch]
 
+        if not IPloneSiteRoot.providedBy(self.context):
+            storage = [
+                path for path in storage._paths.items() if path[1][0] == context_path
+            ]
+
+        query = form.get("q")
+        if query and query.startswith("/"):
+            min_k = f"{portal_path}/{query.strip('/')}"
+            max_k = min_k[:-1] + chr(ord(min_k[-1]) + 1)
+            redirects = storage._paths.keys(min=min_k, max=max_k, excludemax=True)
+        elif query:
+            redirects = [path for path in storage._paths.keys() if query in path]
+        else:
+            redirects = storage._paths.keys()
```

Comment:
If we're not on a root item, we need to filter down the redirects from above, instead of starting with the full set of redirects from the storage.